win32ss/drivers/videoprt/videoprt.c

```diff
@@ -236,17 +295,43 @@ IntVideoPortCreateAdapterDeviceObject(
     KeInitializeMutex(&DeviceExtension->DeviceLock, 0);
 
     /* Attach the device. */
-    if (PhysicalDeviceObject != NULL)
+    if ((PhysicalDeviceObject != NULL) && (DisplayNumber == 0))
         DeviceExtension->NextDeviceObject = IoAttachDeviceToDeviceStack(
                                                 *DeviceObject,
                                                 PhysicalDeviceObject);
 
-    IntCreateNewRegistryPath(DeviceExtension);
+    Status = IntCreateNewRegistryPath(DeviceExtension);
+    if (!NT_SUCCESS(Status))
+    {
+        ERR_(VIDEOPRT, "IntCreateNewRegistryPath() failed with status 0x%08x\n", Status);
+        IoDeleteDevice(*DeviceObject);
+        *DeviceObject = NULL;
+        return Status;
+    }
+
     IntSetupDeviceSettingsKey(DeviceExtension);
-    DriverExtension->InitializationData.StartingDeviceNumber++;
 
     /* Remove the initailizing flag */
     (*DeviceObject)->Flags &= ~DO_DEVICE_INITIALIZING;
+
+    /* Update MaxObjectNumber */
+    VideoPortMaxObjectNumber = DeviceNumber;
```

Comment:
I just realized that in case the below function fails, this would not be correct anymore. A solution might be to assign this in IntVideoPortAddDeviceMapLink, or add another parameter for MaxDeviceNumber, and set it here only on success. But I don't have a strong opinion on it and I think it's not a big issue.

Reply:
Moved VideoPortMaxObjectNumber assignation to the end of IntVideoPortAddDeviceMapLink